src/GHC/Util.hs

```diff
@@ -288,3 +297,25 @@ getloc = getLoc
 
 noext :: NoExt
 noext = noExt
+
+--
+
+newtype SrcSpanD = SrcSpanD SrcSpan -- SrcSpan, deriving `Default`.
+instance Default SrcSpanD where def = SrcSpanD noSrcSpan
+instance Outputable SrcSpanD where ppr (SrcSpanD s) = ppr s
+instance Eq SrcSpanD where (==) (SrcSpanD a) (SrcSpanD b) = a == b
+instance Ord SrcSpanD where compare (SrcSpanD a) (SrcSpanD b) = compare a b
+
+newtype W a = W a -- Wrapper of terms.
```

Comment:
Not sure I like that name `W`. Any better ideas? I don't have anything immediate...

Reply:
I went through various incantations involving `Wrap`. In the end I forsee this as being somewhat ubiquitous and reason short is good.

Reply:
OK, fair enough.